lib/model/recent_senders.dart

```diff
@@ -0,0 +1,189 @@
+import 'dart:math' as math;
+
+import '../api/model/model.dart';
+
+extension Max<T extends num> on Iterable<T> {
+  /// Finds the maximum number in an [Iterable].
+  ///
+  /// Returns null if the [Iterable] is empty.
+  T? max() => isNotEmpty ? reduce(math.max) : null;
+}
+
+extension PossibleIndex<T extends num> on List<T> {
+  /// The index of [element] in this sorted list.
+  ///
+  /// Uses binary search to find the index, so the list must be sorted,
+  /// otherwise the result is unpredictable.
+  ///
+  /// If [element] is found, returns its index (0-based).
+  ///
+  /// ```dart
+  /// final somePrimes = [11, 13, 17, 23, 27, 29];
+  /// final index = somePrimes.possibleIndexOf(17); // 2
+  /// ```
+  /// If [element] is not found, returns the possible index (1-based)
+  /// with a negative sign, where if the [element] is inserted,
+  /// the order of the list is maintained.
+  ///
+  /// ```dart
+  /// final somePrimes = [11, 13, 17, 23, 27, 29];
+  /// int index = somePrimes.possibleIndexOf(7); // -1
+  /// index = somePrimes.possibleIndexOf(19); // -4
+  /// ```
+  int possibleIndexOf(T element) {
+    int low = 0;
+    int high = length - 1;
+
+    while (high >= low) {
+      int mid = (low + high) ~/ 2;
+      if (element == this[mid]) {
+        return mid;
+      } else if (element < this[mid]) {
+        high = mid - 1;
+      } else {
+        low = mid + 1;
+      }
+    }
+    return -low - 1;
```

Comment:
Using `-low - 1` instead of just `-low` will differentiate between the case where the element is found at _**index 0**_ and the case where the element is not found, but should be placed at the very start of the list.

Reply:
Let's explain this in a code comment. 🙂 It's exactly the kind of thing that any reader will be interested in, and most readers will just be looking at the code, without also reading through this PR thread.